mobile/src/utils/connection.js

```diff
@@ -0,0 +1,30 @@
+import { Platform, NetInfo } from 'react-native';
+import { setConnectionState } from '../actions';
+import { dispatch } from '../App';
+
+const isIos = Platform.OS === 'ios';
+
+export function setConnection(connInfo) {
+    dispatch(setConnectionState(connInfo));
+}
+
+export async function retryIfNotConnected(connInfo) {
+  if (connInfo.type === 'none' || connInfo.type === 'unknown') {
+    NetInfo.getConnectionInfo().then((connectionInfo) => {
+      return connInfo;
+    });
+  }
+  return connInfo;
+}
+
+export function handleConnectivityChange(connInfo) {
+  if (isIos) {
+    // retry the connection check twice on ios
+    // https://github.com/facebook/react-native/issues/8615
```

Comment:
https://github.com/facebook/react-native/issues/8615 this was most likely the cause